### ISSUE TITLE
Support for react-like .tsx function components

### ISSUE THREAD

**Issue by danyadev** [vuejs/eslint-plugin-vue#1987]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.22.0
- **eslint-plugin-vue version:** 9.5.1
- **Node version:** 14.18.0
- **Operating System:** macOS Monterey 12.3.1 arm64

**Please show your full configuration:**

```json5
{
  root: true,
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2021,
    project: './tsconfig.json'
  },
  env: {
    browser: true,
    node: true,
    es2021: true
  },
  extends: [
    'plugin:vue/vue3-recommended',
    'plugin:import/typescript',
    'plugin:@typescript-eslint/recommended'
  ],
  // ...
}
```

**What did you do?**	
```tsx
import './App.css'
import { defineComponent } from 'vue'
import { Titlebar } from 'ui/app/Titlebar/Titlebar'

export const App = defineComponent(() => {
  return () => (
    <div class="App" data-scheme="bright_light">
      <Titlebar />
    </div>
  )
})
```

[little more complex example](https://github.com/danyadev/vk-desktop/blob/netflixAdaptation/src/ui/app/Titlebar/Titlebar.tsx)

**What actually happened?**

This is not really a bug report or feature request, just a question:
Is there any support for such components in this plugin and is it worth bothering with its configuration?
